engine/src/main/java/org/terasology/world/block/IBlockRegion.java

```diff
@@ -0,0 +1,145 @@
+import org.joml.LineSegmentf;
+import org.joml.Matrix4fc;
+import org.joml.Planef;
+import org.joml.Rayf;
+import org.joml.Spheref;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.entitySystem.entity.EntityRef;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+public interface IBlockRegion {
+    public static BlockRegion invalid();
+
+    // -- CONSTRUCTORS ----------------------------------------------------------------------------------------------//
+    BlockRegion();
+    public BlockRegion(int minX, int minY, int minZ, int maxX, int maxY, int maxZ);
+    public BlockRegion(Vector3ic min, Vector3ic max);
+    public BlockRegion(int x, int y, int z);
+    public BlockRegion(Vector3ic block);
+    public BlockRegion(BlockRegion source)
+
+    // -- ITERABLE -------------------------------------------------------------------------------------------------//
+    public Iterator<Vector3fc> iterator();
+
+    // -- GETTERS & SETTERS -----------------------------------------------------------------------------------------//
+    public BlockRegion set(int minX, int minY, int minZ, int maxX, int maxY, int maxZ);
+    public BlockRegion set(Vector3ic min, Vector3ic max);
+    public BlockRegion set(BlockRegion source);
+
+    // -- min -------------------------------------------------------------------------------------------------------//
+    public int minX();
+    public BlockRegion minX(int x);
+
+    public int minY();
+    public BlockRegion minY(int y);
+
+    public int minZ();
+    public BlockRegion minZ(int z);
+
+    public Vector3i getMin(Vector3i dest);
+
+    public BlockRegion setMin(int x, int y, int z);
+    public BlockRegion setMin(Vector3ic min);
+    public BlockRegion translateMin(int dx, int dy, int dz);
+    public BlockRegion translateMin(Vector3ic dmin);
+
+    // -- max -------------------------------------------------------------------------------------------------------//
+    public int maxX();
+    public BlockRegion maxX(int x);
+
+    public int maxY();
+    public BlockRegion maxY(int y);
+
+    public int maxZ();
+    public BlockRegion maxZ(int z);
+
+    public Vector3i getMax(Vector3i dest);
+
+    public BlockRegion setMax(int x, int y, int z);
+    public BlockRegion setMax(Vector3ic max);
+    public BlockRegion translateMax(int dx, int dy, int dz);
+    public BlockRegion translateMax(Vector3ic dmax);
+
+    // -- size ------------------------------------------------------------------------------------------------------//
+    public int getSizeX();
+    public int getSizeY();
+    public int getSizeY();
+    public Vector3i getSize(Vector3i dest);
+
+    public BlockRegion setSize(int sizeX, int sizeY, int sizeZ);
+    public BlockRegion setSize(Vector3ic size);
+
+    public int size();                                              // what to call this? size, volume, blockCount, ...
```

Comment:
This question still holds - what should this be called?

Reply:
Volume is more descriptive then size.